File: rl_starter_11/test_castling.py
import sys
sys.path.append("/Users/Shared/西洋棋代理人/rl_starter_11/agents/d6_cpp")
import chess_engine

print("Move action for e8g8 (black FEN):")
print(chess_engine.test_move_to_action("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b KQkq - 0 1", "e8g8"))

File: rl_starter_11/test_castling.cpp
#include <iostream>
#include "chess.hpp"

using namespace chess;

int main() {
    Board board("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1");
    Move m = uci::uciToMove(board, "e1g1");
    std::cout << "UCI e1g1 -> type: " << (int)m.typeOf() << " from: " << (int)m.from().index() << " to: " << (int)m.to().index() << std::endl;
    return 0;
}
